gmso/tests/test_topology.py

```diff
@@ -418,4 +418,45 @@ def test_topology_atom_type_changes(self):
         assert top.sites[10].atom_type.name == 'atom_type_changed'
         assert top.is_typed()
 
+    def test_topology_get_index(self):
+        top = Topology()
+        conn_members = [Site(), Site(), Site(), Site()]
+        for i in range(5):
+            top.add_site(Site())
+            top.add_connection(Bond(
+                connection_members=[conn_members[0], conn_members[1]]))
+            top.add_connection(Angle(
+                connection_members=[conn_members[0], conn_members[1], conn_members[2]]))
+            top.add_connection(Dihedral(
+                connection_members=[conn_members[0], conn_members[1], conn_members[2], conn_members[3]]))
+            top.add_connection(Improper(
+                connection_members=[conn_members[0], conn_members[1], conn_members[2], conn_members[3]]))
+        a_bond = Bond(connection_members=[conn_members[0], conn_members[1]])
+        an_angle = Angle(connection_members=[conn_members[0], conn_members[1], conn_members[2]])
+        a_site = Site()
+        a_dihedral = Dihedral(connection_members=[conn_members[0], conn_members[1], conn_members[2], conn_members[3]])
+        an_improper = Improper(connection_members=[conn_members[0], conn_members[1], conn_members[2], conn_members[3]])
+
+        top.add_site(a_site)
+        top.add_connection(a_bond)
+        top.add_connection(an_angle)
+        top.add_connection(a_dihedral)
+        top.add_connection(an_improper)
+
+        assert top.get_index(a_site) == 9
```

Comment:
Just to make sure i am properly understanding. We add those 5 sites in the `for` loop, and when we create a connection member, if those sites are not in the topology to begin with, those are added too?

So in this case, we add 5 sites in the for loop `0`, and then `1,2,3,4` are from the connection members, then the additional 4 sites from the rest of the for loop: `5,6,7,8`

and then `a_site` at `9`?

I can understand that, but I am confused about the connection members below.

Does this:
```python3
            top.add_connection(Angle(
                connection_members=[conn_members[0], conn_members[1], conn_members[2]]))
```
Not just keep adding the same connection over and over again, but since it is the same, its just added once?
I would assume that the index for each connection type after the for loop would only have `0`, just the single one.

Then a second one after the for loop?

Reply:
https://github.com/mosdef-hub/gmso/blob/master/gmso/core/topology.py#L373-L375
Since the internal representation is an IndexedSet. Only one instance of a `Site` can exist in the topology.